includes/service-providers/constant_contact/class-newspack-newsletters-constant-contact.php

```diff
@@ -825,4 +832,111 @@ public static function get_labels() {
 		);
 	}
 
+	/**
+	 * Retrieve the ESP's tag ID from its name
+	 *
+	 * @param string  $tag_name The tag.
+	 * @param boolean $create_if_not_found Whether to create a new tag if not found. Default to true.
+	 * @param string  $list_id The List ID.
+	 * @return int|WP_Error The tag ID on success. WP_Error on failure.
+	 */
+	public function get_tag_id( $tag_name, $create_if_not_found = true, $list_id = null ) {
+		$cc  = new Newspack_Newsletters_Constant_Contact_SDK( $this->api_key(), $this->api_secret(), $this->access_token() );
```

Comment:
Non-blocking nitpick since it doesn't seem like it will result in any real performance savings, but should we create and cache a single instance of the SDK class instead of creating a new instance each request?

Reply:
YES! I thought the same thing, but I just followed the pattern there and didn't engage in a refactor to not inflate the PR.

I added a comment about this in [this issue](https://github.com/Automattic/newspack-newsletters/issues/1022), so we remember of it when we do some needed refactoring.